src/elements/content-picker/ContentPicker.js

```diff
@@ -820,6 +827,55 @@ class ContentPicker extends Component<Props, State> {
         });
     };
 
+    /**
+     * Show the inline shared link dropdown
+     * @param {BoxItem} item - The item (folder, file, weblink)
+     * @returns {Promise}
+     */
+    showSharedLinkDropdown = (item: BoxItem): void => {
+        const { id, type }: BoxItem = item;
+
+        switch (type) {
+            case TYPE_FOLDER:
+                this.api.getFolderAPI().getFolderFields(id, this.handleSharedLinkResponse, () => {}, {
+                    fields: FILE_SHARED_LINK_FIELDS_TO_FETCH,
+                });
+                break;
+            case TYPE_FILE:
+                this.api
+                    .getFileAPI()
+                    .getFile(id, this.handleSharedLinkResponse, () => {}, { fields: FILE_SHARED_LINK_FIELDS_TO_FETCH });
+                break;
+            case TYPE_WEBLINK:
+                break;
+            default:
+                throw new Error('Unknown Type');
+        }
+    };
+
+    /**
+     * Handles the shared link info by either creating a share link using enterprise defaults if
+     * it does not already exist, otherwise update the item in the state currentCollection.
+     *
+     * @param {Object} item file or folder
+     * @returns {void}
+     */
+    handleSharedLinkResponse = (item: BoxItem) => {
+        // if no shared link currently exists, create a shared link with enterprise default
+        if (!item[FIELD_SHARED_LINK]) {
+            this.changeShareAccess(null, item);
+        } else {
+            const { selected } = this.state;
+            const { id, type } = item;
+            const cacheKey = this.api.getAPI(type).getCacheKey(id);
+            // if shared link already exists, update the collection in state
+            this.updateItemInCollection(item);
+            if (item.selected && item !== selected[cacheKey]) {
```

Comment:
Does this handle the case where the user selects a new item before the shared link response returns? May want to add a comment.

Reply:
Yeah you're right, if the user clicks another item before the XHR completes it gets cancelled. I'll add a comment about that

Reply:
Consider adding a comment.